### PR TITLE
Redirect to login

### PR DESCRIPTION
**Pull request by ben-dh3**
- If user is not logged in (i.e. there is no token in local storage) redirect to login page when generate recipe is clicked. 
- 'handleLogin' function tidied on app.jsx

### CODE REVIEW

**Review by Ro-0171 — APPROVED** [review]

File: frontend/src/components/RecipeScraper.jsx [hunk 1]
```diff
@@ -8,6 +8,16 @@ const RecipeScraper = ({
 }) => {
   const navigate = useNavigate();
 
+  // if the user is logged in - generate a recipe, else redirect to login page
+  const handleClick = async () => {
+    if (localStorage.getItem('token')) {
+      await handleScrapeRecipe();
```
Comment (kawrou) [inline frontend/src/components/RecipeScraper.jsx:16]: Is handleScrapeRecipe() being passed as props to RecipeScraper?
Hard to follow on github. Can't quite track where setToken is being used after getting back the token from handleScrapeRecipe. Though I guess the code atm doesn't return a token from the backend.

Comment (gabrielaehrenbrink) [reply frontend/src/components/RecipeScraper.jsx:16]: Yeah, handleScrapeRecipe is being passed as a prop, not so sure about the token, I added to the backend a generate token for the generate recipe, but that hasn't been merged, before that I think the token wasn't really needed to scrape to recipe (could be wrong, but that was how we first did it)

Comment (kawrou) [reply frontend/src/components/RecipeScraper.jsx:16]: Yeah, It wasn't being used, but probably need to be.
